tests/test_utils.py

```diff
@@ -677,6 +677,13 @@ def QueryValueEx(key, value_name):
             elif value_name == 'ProxyOverride':
                 return [override]
 
+    def QueryValueEx2(key, value_name):
+        if key is ie_settings:
+            if value_name == 'ProxyEnable':
+                return ["1"] # this could be a string (REG_SZ) or a 32-bit number (REG_DWORD)
+            elif value_name == 'ProxyOverride':
+                return [override]
+
```

Comment:
This is not sufficient, as it has never been used. However, you're right that this never actually tests anything, which isn't great. This was supposed to call `should_bypass_proxies`: want to add that call, and then parametrise this test based on the return value of `QueryValueEx` with `ProxyEnable`?

Reply:
Let me try adding some real assertions here. Need to find a Windows first, though.